snap/snapcraft.yaml

```diff
@@ -39,25 +39,7 @@ description: |
   Use this when restarts should be avoided e.g. when running a validator.
 
 parts:
-
-  rust-deps:
-    plugin: nil
-    build-packages:
-     - curl
-    override-pull: |
-      curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
```

Comment:
yes, you are correct this command installs `rustup` and has worked previously when we updated the Polkadot version but for this update, it didn't work. `cargo` wasn't set up properly nor was the `toolchain`.  So other around that we figured out was that we don't need to install the rustup in a separate part of `rust-deps` because it will be done by `plugin` where we have provided `rust`. I am sorry I couldn't find something concrete to support my stance but this is what I found through a discussion with [ChatGPT](https://chat.openai.com/share/431bfa58-6943-4ab1-a617-8c5780a683e2)
CC: @erik78se @Maharacha